src/test/java/hudson/plugins/deploy/MockAdapter.java

```diff
@@ -0,0 +1,44 @@
+package hudson.plugins.deploy;
+
+import hudson.Extension;
+import hudson.FilePath;
+import hudson.Launcher;
+import hudson.model.AbstractBuild;
+import hudson.model.BuildListener;
+import org.jenkinsci.Symbol;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+import java.io.IOException;
+
+/**
+ * Mocks out a ContainerAdapter so that we can actually run the deployer in a trivial way, just to test pipeline compatibility
+ * CHECKME Might need to extend CargoContainerAdapter to better exercise APIs
+ */
+@Extension
+public class MockAdapter extends ContainerAdapter {
```

Comment:
I would call it more explicitly, e.g. `LegacyAbstractProjectOnlyContainerAdapter`